delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala

```diff
@@ -0,0 +1,196 @@
+package ch.epfl.bluebrain.nexus.delta.routes
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.schemas
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
+import ch.epfl.bluebrain.nexus.delta.routes.directives.DeltaDirectives._
+import ch.epfl.bluebrain.nexus.delta.routes.marshalling.CirceUnmarshalling
+import ch.epfl.bluebrain.nexus.delta.routes.marshalling.RdfRejectionHandler._
+import ch.epfl.bluebrain.nexus.delta.sdk.Permissions.{events, resources => resourcePermissions}
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.AuthDirectives
+import ch.epfl.bluebrain.nexus.delta.sdk.error.ServiceError.AuthorizationFailed
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.{IriSegment, StringSegment}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRejection.ProjectNotFound
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ProjectRef, ProjectRejection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, IdSegment, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resources.ResourceRejection
+import io.circe.Json
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+import monix.execution.Scheduler
+
+/**
+  * The resource routes
+  *
+  * @param identities the identity module
+  * @param acls       the ACLs module
+  * @param projects   the projects module
+  * @param resources  the resources module
+  */
+final class ResourcesRoutes(
+    identities: Identities,
+    acls: Acls,
+    projects: Projects,
+    resources: Resources
+)(implicit baseUri: BaseUri, s: Scheduler, cr: RemoteContextResolution, ordering: JsonKeyOrdering)
+    extends AuthDirectives(identities, acls)
+    with CirceUnmarshalling {
+
+  import baseUri.prefixSegment
+
+  private val resourceSchema = IriSegment(schemas.resources)
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractCaller { implicit caller =>
+        pathPrefix("resources") {
+          concat(
+            // SSE resources
+            (pathPrefix("events") & pathEndOrSingleSlash) {
+              get {
+                operationName(s"$prefixSegment/resources/events") {
+                  authorizeFor(AclAddress.Root, events.read).apply {
+                    lastEventId { offset =>
+                      emit(resources.events(offset))
+                    }
+                  }
+                }
+              }
+            },
+            projectRef { ref =>
+              routesForProject(ref)
+            },
+            (uuid & uuid) { (orgUuid, projectUuid) =>
```

Comment:
We could have a directive projectRefFromUuid

Reply:
Yes, but would it make things clearer? I find this line already very explicit :)